docs/user_guide/usage_patterns/data_transfer/detecting_data_access/index.rst

```diff
@@ -0,0 +1,161 @@
+``data_access`` scope.
+
+``data_access`` scopes control application access to collections, allowing
+users to revoke access for an application independent from other application
+permissions.
+Revoking consent stops data transfers and other operations.
+
+Because only some collection types have ``data_access`` scopes, application
+authors interacting with these collections may need to detect the type of
+collection and determine whether or not the scope will be needed.
+
+For readers who prefer to start with complete working examples, jump ahead to the
+:ref:`example script <userguide_detecting_data_access_example>`.
+
+Accessing Collections in Globus Transfer
+----------------------------------------
+
+The Globus Transfer service acts as a central registration hub for collections.
+Therefore, in order to get information about an unknown collection, we will
+need a :class:`TransferClient` with credentials.
+
+The following snippet creates a client and uses it to fetch a collection from
+the service:
+
+.. code-block:: python
+
+    import globus_sdk
+
+    # Tutorial Client ID - <replace this with your own client>
+    NATIVE_CLIENT_ID = "61338d24-54d5-408f-a10d-66c06b59f6d2"
+    USER_APP = globus_sdk.UserApp("detect-data-access-example", client_id=NATIVE_CLIENT_ID)
+
+    # Globus Tutorial Collection 1
+    # https://app.globus.org/file-manager/collections/6c54cade-bde5-45c1-bdea-f4bd71dba2cc
+    # replace with your own COLLECTION_ID
+    COLLECTION_ID = "6c54cade-bde5-45c1-bdea-f4bd71dba2cc"
+
+    transfer_client = globus_sdk.TransferClient(app=USER_APP)
+
+    collection_doc = transfer_client.get_endpoint(COLLECTION_ID)
+
+.. caution::
+
+    Careful readers may note that we the :meth:`TransferClient.get_endpoint`
+    method to lookup a collection.
+
+    The Transfer service contains both Endpoints and Collections, and for
+    historical reasons both document types are available from the Get Endpoint
+    API.
+    This is the correct API to use not only when looking for information about
+    a Collection, but also for detecting whether an ID belongs to a Collection
+    or Endpoint, in cases where it is ambiguous.
+
+
+Reading Collection Type
+-----------------------
+
+There are two attributes we need from the collection document to determine
+whether or not a ``data_access`` scope is used.
+
+First, whether or not the collection is a GCSv5 Mapped Collection:
+
+.. code-block:: python
+
+    entity_type: str = collection_doc["entity_type"]
+    is_v5_mapped_collection: bool = entity_type == "GCSv5_mapped_collection"
+
+Second, whether or not the collection is a High Assurance Collection:
+
+.. code-block:: python
+
+    high_assurance: bool = collection_doc["high_assurance"]
+
+Once we have this information, we can deduce whether or not ``data_access`` is
+needed with the following boolean assignment:
+
+.. code-block:: python
+
+    collection_uses_data_access: bool = is_v5_mapped_collection and not high_assurance
```

Comment:
Et cetera. In general, the code snippets don't need type annotations IMO. We can perhaps have them in the example script, but I don't think we need anything other than the bare code logic throughout here.

```suggestion
    collection_uses_data_access = is_v5_mapped_collection and not high_assurance
```

Reply:
I think we can trust users to do basic deductions that "the outcome of boolean ops on bools is a bool", or "if I compared a string against the field, it's probably a string".

So in this case I'm more open to removal because the type is _knowable_ without prior knowledge of the payload data. However, in the `"high_assurance"` field case, I think indicating the type is important. If we use annotations there, it might be best to be consistent and use them on all vars. If we use a non-annotation way of indicating the type, I would strip all of the annotations (again, for internal consistency).